sdks/java/core/src/test/java/org/apache/beam/sdk/io/FileBasedSinkTest.java

```diff
@@ -198,23 +198,23 @@ private void runFinalize(SimpleSink.SimpleWriteOperation writeOp, List<File> tem
       throws Exception {
     int numFiles = temporaryFiles.size();
 
-    List<FileResult> fileResults = new ArrayList<>();
+    List<FileResult<Void>> fileResults = new ArrayList<>();
     // Create temporary output bundles and output File objects.
     for (int i = 0; i < numFiles; i++) {
       fileResults.add(
-          new FileResult(
+          new FileResult<Void>(
               LocalResources.fromFile(temporaryFiles.get(i), false),
               WriteFiles.UNKNOWN_SHARDNUM,
               null,
+              null,
               null));
     }
 
     writeOp.finalize(fileResults);
 
-    ResourceId outputDirectory = writeOp.getSink().getBaseOutputDirectoryProvider().get();
     for (int i = 0; i < numFiles; i++) {
-      ResourceId outputFilename = writeOp.getSink().getFilenamePolicy()
-          .unwindowedFilename(outputDirectory, new Context(i, numFiles), "");
+      ResourceId outputFilename = writeOp.getSink().getDynamicDestinations().getFilenamePolicy(null)
+          .unwindowedFilename(new Context(i, numFiles), CompressionType.UNCOMPRESSED);
```

Comment:
OutputFileHints is an interface right now. I'm not sure I like the idea of creating an explicit implementation of it just for these tests. In general I'm unhappy with the OutputFileHints, WrieableByteChannel, CompressionType hierarchy, but it was the best compromise I saw that didn't break backwards compatibility.